src/data/logfiltereddataworkerthread.cpp

```diff
@@ -220,41 +236,58 @@ void SearchOperation::doSearch( SearchData& searchData, qint64 initialLine )
     const qint64 nbSourceLines = sourceLogData_->getNbLine();
     int maxLength = 0;
     int nbMatches = searchData.getNbMatches();
-    SearchResultArray currentList = SearchResultArray();
+    SearchResultArray currentList;
+
+    std::vector<NumberedLine> numberedLines;
 
     // Ensure no re-alloc will be done
     currentList.reserve( nbLinesInChunk );
+    numberedLines.reserve( nbLinesInChunk );
 
     LOG(logDEBUG) << "Searching from line " << initialLine << " to " << nbSourceLines;
 
-    for ( qint64 i = initialLine; i < nbSourceLines; i += nbLinesInChunk ) {
+    for ( qint64 chunkStart = initialLine; chunkStart < nbSourceLines; chunkStart += nbLinesInChunk ) {
         if ( *interruptRequested_ )
             break;
 
-        const int percentage = ( i - initialLine ) * 100 / ( nbSourceLines - initialLine );
+        const int percentage = ( chunkStart - initialLine ) * 100 / ( nbSourceLines - initialLine );
         emit searchProgressed( nbMatches, percentage );
 
-        const QStringList lines = sourceLogData_->getLines( i,
-                qMin( nbLinesInChunk, (int) ( nbSourceLines - i ) ) );
-        LOG(logDEBUG) << "Chunk starting at " << i <<
+        const int linesInChunk = qMin( nbLinesInChunk, (int) ( nbSourceLines - chunkStart ) );
+        const QStringList lines = sourceLogData_->getLines( chunkStart, linesInChunk );
+
+        LOG(logDEBUG) << "Chunk starting at " << chunkStart <<
             ", " << lines.size() << " lines read.";
 
-        int j = 0;
-        for ( ; j < lines.size(); j++ ) {
-            if ( regexp_.indexIn( lines[j] ) != -1 ) {
-                // FIXME: increase perf by removing temporary
-                const int length = sourceLogData_->getExpandedLineString(i+j).length();
```

Comment:
Looks like getExpandedLineString is reading string from file and replaces tabs with spaces. If this is right than the same process can be done with line[j] directly. And as only length is needed here I decided to introduce a function which does just expected untabified string length calculation